src/main/webapp/app/components/tokenInputGroups/TokenExtendButton.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { ReactNode } from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row
+} from 'react-bootstrap';
+import classnames from 'classnames';
+import { observable, action } from 'mobx';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+
+type TokenExtendButtonProps = {
+  currentDate: string;
+  afterChangeDate: (newDate: string) => void;
+};
+
+@observer
+export default class TokenExtendButton extends React.Component<
+  TokenExtendButtonProps
+> {
+  @observable selectedDay: Date;
+
+  constructor(props: TokenExtendButtonProps) {
+    super(props);
+  }
+
+  @autobind
+  @action
+  handleDayChange(day: any) {
+    this.selectedDay = day;
+  }
+
+  @autobind
+  @action
+  handleSelectedDay() {
+    this.props.afterChangeDate(
+      moment(this.selectedDay).format(APP_LOCAL_DATETIME_FORMAT_Z_FORCE)
+    );
+  }
+
+  render() {
+    const extendSchedule = [
+      {
+        amount: 1,
+        idx: 1,
+        unit: 'week'
+      },
+      {
+        amount: 1,
+        idx: 2,
+        unit: 'month'
+      },
+      {
+        amount: 6,
+        idx: 3,
+        unit: 'months'
+      },
+      {
+        amount: 1,
+        idx: 4,
+        unit: 'year'
+      }
+    ];
+    return (
+      <DefaultTooltip
+        placement={'top'}
+        overlay={'Extend token expiration date'}
+      >
+        <DropdownButton
+          as={InputGroup.Append}
+          title={<i className={classnames('fa fa-calendar')}></i>}
+          id="time-entend-dropdown"
+        >
+          {extendSchedule.map(val => {
+            let unit: moment.unitOfTime.DurationConstructor;
+            switch (val.unit) {
+              case 'day':
+              case 'days':
+                unit = 'd';
+                break;
+              case 'week':
+              case 'weeks':
+                unit = 'w';
+                break;
+              case 'month':
+              case 'months':
+                unit = 'M';
+                break;
+              case 'year':
+              case 'years':
+                unit = 'y';
+                break;
+              default:
+                unit = 'd';
+                break;
```

Comment:
Why don't we just add one more property in the `extendSchedule`?

Reply:
Cause the unit needs to be declared as `moment.unitOfTime.DurationConstructor` or it cannot be added by `Moment.add`.